common/planners/src/plan_node.rs

```diff
@@ -92,14 +99,25 @@ pub enum PlanNode {
 
     // Insert.
     Insert(InsertPlan),
+
+    // Copy.
     Copy(CopyPlan),
-    Sink(SinkPlan),
+
+    // Show.
+    ShowDatabases(ShowDatabasesPlan),
```

Comment:
Can we merge them into one show plan ?

This makes plan enum struct much more fatter.

Reply:
> This makes plan enum struct much more fatter.

For sql statement, it looks we should transform the statement to the deterministic plan, the plan will transformed to the deterministic interpreter, each interpreter will has his own privileges check.
`show databases` -- database level check, some no privileges database will not show here
`show tables` -- table level check

Reply:
Can we apply the ` privileges check` inside the show plans?  `Show(ShowPlan)`

```
struct ShowPlan {
  Kind: Kind
}
```

Reply:
> Can we apply the ` privileges check` inside the show plans? `Show(ShowPlan)`
> 
> ```
> struct ShowPlan {
>   Kind: Kind
> }
> ```

Yes, sound a good idea.
The current implemation is the flat version of the kind.

Now, there are only 8 `show []`:
* show databases
* show tables
* show functions
* show users
* show stages -- TODO
* show engines -- management mode disabled
* show metrics -- management mode disabled
* show processlist -- management mode disabled
If we bring in the `kind`, it looks we back to the `match kind` and do every query rewritten in one place.